aws-sdk-resources/lib/aws-sdk-resources/services/s3/encryption/io_auth_decrypter.rb

```diff
@@ -0,0 +1,52 @@
+module Aws
+  module S3
+    module Encryption
+      # @api private
+      class IOAuthDecrypter
+
+        # @option options [required, IO#write] :io
+        #   An IO-like object that responds to {#write}.
+        # @option options [required, Integer] :encrypted_content_length
+        #   The number of bytes to decrypt from the `:io` object.
+        #   This should be the total size of `:io` minus the length of
+        #   the cipher auth tag.
+        # @option options [required, OpenSSL::Cipher] :cipher
```

Comment:
Can we document these two options, even if very succinctly? Very helpful down the line when debugging and needed to reload context.